include/snowflake/sb_CRTFunctionSafe.h

```diff
@@ -21,186 +17,257 @@
 extern "C" {
 #endif
 
-#ifndef STDCALL
-    // match snowflake define in platform.h
-    #ifdef _WIN32
-        #define STDCALL __stdcall
-    #else
-        #define STDCALL
-    #endif
-#endif
-
-extern char* STDCALL sf_getenv(const char *name);
-extern char* STDCALL sf_strerror(int);
+#ifndef STDCALL
+    // match snowflake define in platform.h
+    #ifdef _WIN32
+        #define STDCALL __stdcall
+    #else
+        #define STDCALL
+    #endif
+#endif
+
+extern char* STDCALL sf_getenv(const char *name, char *outbuf, size_t bufsize);
+extern char* STDCALL sf_strerror(int errnum, char* outbuf, size_t bufsize);
+
+    /// @brief Copy bytes between buffers.
+    ///
+    /// @param out_dest         Destination buffer. (NOT OWN)
+    /// @param in_destSize      Size of the destination buffer.
+    /// @param in_src           Buffer to copy from. (NOT OWN)
+    /// @param in_sizeToCopy    Number of bytes to copy.
+    ///
+    /// @return A pointer to destination; NULL if an error occurred. (NOT OWN)
+    static inline void* sb_memcpy(
+        void* out_dest,
+        size_t in_destSize,
+        const void* in_src,
+        size_t in_sizeToCopy)
+    {
+#if defined(_WIN32) || defined(_WIN64)
+        if (0 == memcpy_s(out_dest, in_destSize, in_src, in_sizeToCopy))
+        {
+            return out_dest;
+        }
+        return NULL;
+#else
+        if (in_sizeToCopy > in_destSize)
+        {
+            return NULL;
+        }
+        return memcpy(out_dest, in_src, in_sizeToCopy);
+#endif
+    }
 
-#if defined(WIN32) || defined(_WIN64)   // For Windows
+    // helper function for sb_strcpy, sb_strncpy and sbcat on non-Windows
+    static inline char* sb_copy(void* dst, size_t dstsize, char const* src, long long srclen)
+    {
+        const size_t copyLen = (srclen < 0) ? strlen(src) + 1 : srclen;
 
-    #define sf_free_s(x)        free(x)
+        //NOTE: this copies the terminul:
+        return (char*)sb_memcpy(dst, dstsize, src, copyLen);
+    }
+
+    // helper function for sb_strcat and sb_strncat
+    static inline char* sb_cat(char* dst, size_t dstsize, char const* src, size_t srclen)
+    {
+        size_t dstlen = strlen(dst);
+        return dstsize < dstlen ?
+            NULL :
+            sb_copy(dst + dstlen, dstsize - dstlen,
+                        src, srclen < 0 ? strlen(src) + 1 : srclen);
+    }
 
-    /// @brief Copy a string.
-    /// 
+/// @brief Copy a string.
+    ///
     /// @param out_dest         Destination string buffer. (NOT OWN)
     /// @param in_destSize      Size of the destination string buffer.
     /// @param in_src           Null-terminated source string buffer. (NOT OWN)
-    /// 
+    ///
     /// @return The destination string; NULL if an error occurred. (NOT OWN)
-    inline char* sb_strcpy(
+    static inline char* sb_strcpy(
```

Comment:
should it be `sf_` or `sb_`? it's also the question for other places in this file